### PR TITLE
Handle missing archived index

### LINKED ISSUE #5898: unknown block when requesting historical state
Reported by @0xKiwi, RPC failed with the following errors when fetching historical attestations:

```
[2020-05-17 15:23:30] ERROR detection: Could not fetch attestations for epoch: 458 error=rpc error: code = Internal desc = Could not retrieve state for attestation data block root 0xee59773b78896f046ebe2a6d59f485700754ec74c8d9216b441559b855203c90: unknown block
could not request indexed attestations for epoch: 458
github.com/prysmaticlabs/prysm/slasher/beaconclient.(*Service).RequestHistoricalAttestations
    slasher/beaconclient/historical_data_retrieval.go:32
github.com/prysmaticlabs/prysm/slasher/detection.(*Service).detectHistoricalChainData
    slasher/detection/service.go:124
runtime.goexit
    GOROOT/src/runtime/asm_amd64.s:1373
[2020-05-17 15:23:30] ERROR detection: Could not fetch attestations for epoch: 459 error=rpc error: code = Internal desc = Could not retrieve state for attestation data block root 0xa3b5313c52759218a23c0696c7c9b78d228f215c1329cbcbdc32c65ae0cd1b3b: unknown block
could not request indexed attestations for epoch: 459
github.com/prysmaticlabs/prysm/slasher/beaconclient.(*Service).RequestHistoricalAttestations
    slasher/beaconclient/historical_data_retrieval.go:32
github.com/prysmaticlabs/prysm/slasher/detection.(*Service).detectHistoricalChainData
    slasher/detection/service.go:124
runtime.goexit
    GOROOT/src/runtime/asm_amd64.s:1373
```

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**
> Bug fix

**What does this PR do? Why is it needed?**
When an archived slot has a skip block, the node should be looking back to last archived slot to get a real root instead of returning `0x0000...`This also handles genesis case instead of returning `0x0000...`

**Which issues(s) does this PR fix?**

Fixes #5898

**Other notes for review**
Tested locally. Was able to request previously failed historical states via debug end point

### CODE REVIEW

File: beacon-chain/state/stategen/replay.go [hunk 1]
```diff
@@ -308,18 +308,36 @@ func (s *State) genesisRoot(ctx context.Context) ([32]byte, error) {
...
-// This retrieves the archived root in the DB.
-func (s *State) archivedRoot(ctx context.Context, slot uint64) [32]byte {
+// This retrieves the highest archived root that represents the input slot in the DB.
+func (s *State) archivedRoot(ctx context.Context, slot uint64) ([32]byte, error) {
 	archivedIndex := uint64(0)
 	if slot/params.BeaconConfig().SlotsPerArchivedPoint > 1 {
 		archivedIndex = slot/params.BeaconConfig().SlotsPerArchivedPoint - 1
 	}
-	return s.beaconDB.ArchivedPointRoot(ctx, archivedIndex)
+	fmt.Println(archivedIndex)
```
Comment (0xKiwi) [inline beacon-chain/state/stategen/replay.go:317]: I see this number increasing constantly, reached 130 and I turned off my slasher.

Comment (terencechain) [reply beacon-chain/state/stategen/replay.go]: That's good. That means your slasher is doing work 👍 I removed the debug log

File: beacon-chain/state/stategen/replay.go [hunk 2]
```diff
@@ -308,18 +308,36 @@ func (s *State) genesisRoot(ctx context.Context) ([32]byte, error) {
...
-	return s.beaconDB.ArchivedPointRoot(ctx, archivedIndex)
+
+	for archivedIndex > 0 {
+		if ctx.Err() != nil {
+			return [32]byte{}, ctx.Err()
+		}
+		if s.beaconDB.HasArchivedPoint(ctx, archivedIndex) {
+			return s.beaconDB.ArchivedPointRoot(ctx, archivedIndex), nil
+		}
+		archivedIndex--
```
Comment (prestonvanloon) [inline beacon-chain/state/stategen/replay.go:325]: What is this? Can you add more to the godoc description about what this implementation is doing?

Comment (terencechain) [reply beacon-chain/state/stategen/replay.go:325]: will do

**Pull request merged**